### ISSUE TITLE
channel name requirements

### ISSUE THREAD

**Issue by alexbaden** [openconnectome/ndio#235]
In the `create_channel` function in the neurodata remote (`remote/neurodata.py`), we have the following:

```
   1161             if not c.isalnum():
-> 1162                 raise ValueError("Name cannot contain character {}.".format(c))
```

Thoughts on creating an exceptions list for certain characters that are allowed (and heavily used). The `_` character comes to mind... 

@Aeusman / @kunallillaney Feel free to chime in!

**Comment by Aeusman** [comment]
Duplicate of https://github.com/neurodata/ndstore/issues/21 ?

**Comment by alexbaden** [comment]
Related. ndstore/#21 is primarily about stripping out "-" and delivering an error message in the django UI. This issue is about allowing certain characters ("_"). Personally, I'd rather see failures happen locally (in the UI or in ndio) rather than trying to decode an error message from the server, even if that means we have two separate lists of allowed characters. They likely won't ever change...

**Comment by j6k4m8** [comment]
I agree — no reason to waste time with a REST request if we "know" it'll fail (for some value of _know_). If the rules are as explicit as "alphanumeric with underscores", we can just add that in as an optional (default-on) sanity-check decorator (like `@check_tokens`).

**Comment by Aeusman** [comment]
"alphanumeric with underscores" is correct,  recommended for matching a bad string regex is "^[^$&+,:;=?@#|'<>.^*()%!-]+$"

**Comment by alexbaden** [comment]
Simplest possible thing: 

change `create_channel` from 

```
        for c in name:
            if not c.isalnum():
                raise ValueError("Name cannot contain character {}.".format(c))
```

to 

```
        for c in name:
            if not c.isalnum() and c != "_":
                raise ValueError("Name cannot contain character {}.".format(c))
```

thoughts?

**Comment by wrgr** [comment]
@alexbaden: we intentionally made things as simple as possible to reduce maintenance of the ndio code base since everyone is pretty busy.  I have historically voted against client side code like this in favor of keeping things fairly simple and more in spirit with RESTful wrappers to make it easy to port and manage the code.  This seems like not worth the overhead in general.  I think should be handled server side but just my 2c.

**Comment by alexbaden** [comment]
Sure, but the problem here is ndio is being too restrictive. `_` is an allowed character in channel names. In fact, it is frequently used as a spacer. My proposal is to loosen the ndio restriction to alphanumerics + `_`. We could move this functionality entirely onto the server, but channel naming conventions don't change very much, `_` will pretty much always be an allowed character, and other special characters are not really necessary or not allowed.

**Comment by wrgr** [comment]
meh - I read this incompletely on my phone last night.  This is a great update. Thanks!

**Comment by alexbaden** [comment]
Will be resolved in 1.1.6
